### PR TITLE
Load git objects into datastructure

### LINKED ISSUE #35: Load git objects into datastructure
- [x] Parse main commit
- [x] Parse trees recursively
- [x] Parse blobs

### LINKED ISSUE #49: Test new parser implementation against frontend

### PR DESCRIPTION
**Pull request by joglr**
- [x] Closes #35 
- [x] Closes #49

### CODE REVIEW

**Comment by joglr** [comment]
I'm having some issues where the git object parsing doesn't always work.

**Comment by joglr** [comment]
image
works with parser

**Comment by hojelse** [comment]
Branch specifier crashes.

branch_specifier

File: parser/src/util.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+export function runProcess(command: string, args: string[]) {
+  return new Promise((resolve, reject) => {
+    const prcs = spawn(command, args)
+    prcs.stdout.once("data", (data) => resolve(data.toString()))
+    prcs.stderr.once("data", (data) => reject(data.toString()))
+  })
+}
+
+export async function writeCommitObjectToFile(
```
Comment (tjomson) [inline parser/src/util.ts:14]: Should this be renamed to something like "writeRepoToFile"?
It start from the head commit, but i think the name is a bit misleading, as the function leads to everything being printed (including trees and such). Or have i misunderstood the function?

**Comment by hojelse** [comment]
This would be nice to have #50

**Review by tjomson — APPROVED** [review]
Can expand-tree.ts be deleted now?

**Pull request merged**